tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/ant/CleanOutputDirectoryTask.java

```diff
@@ -100,8 +100,8 @@ public void init() throws BuildException {
     }
 
     @SuppressWarnings("unused")
-    public void setRoot(Path root) {
-        this.root = root;
+    public void setRoot(String root) {
```

Comment:
I understand the charset change (sorry, that was very lax of me).

However I'm interested to know why the Path -> String changes were made, since unless I know what's going on I'm inevitably going to make the same mistake again (since I cannot test on Windows).

Can you add a one-line comment here (and in the other places where you've made changes) explaining it please?

E.g.
// We can't use Path as a parameter type because <insert reason here...>

I assume it's an Ant version thing, not a Windows thing per se, but I could be wrong.

Reply:
Thanks for taking a look!

Right, this is an Ant issue, although I'm unsure if it's due to a different version. Ant doesn't seem to perform conversions for `java.nio.file.Path` objects. Attempting to build results in the following error:

`build-icu-data.xml:339: outputDirectories doesn't support the "root" attribute`

See "Conversions Ant will perform for attributes" here: https://ant.apache.org/manual/develop.html

I'll push an update to add comments explaining this.

Reply:
Maybe say " ...because on some systems Ant doesn't ..."
That way, if someone does try it and it works for them they won't think they can "fix" it.
If you want to keep it on one line, you can probably just have something like:
"Use String here since on some systems Ant doesn't support automatically converting Path instances."